### PR TITLE
feat: ingestion logs and migrations

### PR DESCRIPTION
**Pull request by mfiedorowicz**
- add migrations
- store and retrieve ingestions logs (using `protojson` for consistent bi-directional marshalling/unmarshalling from/to proto)
- improve apply change set error handling

### CODE REVIEW

File: diode-server/reconciler/config.go [hunk 1]
```diff
@@ -2,12 +2,13 @@ package reconciler
...
-	RedisPassword string `envconfig:"REDIS_PASSWORD" required:"true"`
-	RedisDB       int    `envconfig:"REDIS_DB" default:"0"`
-	RedisStreamDB int    `envconfig:"REDIS_STREAM_DB" default:"1"`
+	GRPCPort         int    `envconfig:"GRPC_PORT" default:"8081"`
+	RedisHost        string `envconfig:"REDIS_HOST" default:"127.0.0.1"`
+	RedisPort        string `envconfig:"REDIS_PORT" default:"6379"`
+	RedisPassword    string `envconfig:"REDIS_PASSWORD" required:"true"`
+	RedisDB          int    `envconfig:"REDIS_DB" default:"0"`
+	RedisStreamDB    int    `envconfig:"REDIS_STREAM_DB" default:"1"`
+	MigrationEnabled bool   `envconfig:"MIGRATION_ENABLED" default:"true"`
```
Comment (leoparente) [inline diode-server/reconciler/config.go:11]: I think we should add this new env to sample.env and readme

**Comment by leoparente** [comment]
Entity still not being parsed, right?

**Review by leoparente — APPROVED** [review]
Overall, LGTM. Great work with migrations!

**Comment by mfiedorowicz** [comment]
> Entity still not being parsed, right?

Added appropriate assertions in tests 🔬

**Pull request merged**